src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -25,94 +53,1321 @@
+        sectionPanel.getParticipantPanels().put(currentTeam, teamsMembersPanels);
+    }
+
+    /**
+     * 
+     * 
+     * @param sectionPanel
+     * @param questions
+     * @param responsesGroupedByTeam
+     * @param teamsInSection
+     */
+    private void buildTeamsStatisticsTableForSectionPanel(
+                     InstructorFeedbackResultsSectionPanel sectionPanel, 
+                     LinkedHashMap<String, Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>> responsesGroupedByTeam,
+                     Collection<String> teamsInSection) {
+        Map<String, List<InstructorResultsQuestionTable>> teamToStatisticsTables = new HashMap<String, List<InstructorResultsQuestionTable>>();
+        for (String team : teamsInSection) {
+            if (!responsesGroupedByTeam.containsKey(team)) {
+                continue;
+            }
+            
+            List<InstructorResultsQuestionTable> statisticsTablesForTeam = new ArrayList<InstructorResultsQuestionTable>();
+            
+            for (FeedbackQuestionAttributes question : bundle.questions.values()) {
+                if (!responsesGroupedByTeam.get(team).containsKey(question)) {
+                    continue;
+                }
+                
+                List<FeedbackResponseAttributes> responsesGivenTeamAndQuestion = responsesGroupedByTeam.get(team).get(question);
+        
+                InstructorResultsQuestionTable statsTable = buildQuestionTableWithoutResponseRows(question, 
+                                                                               responsesGivenTeamAndQuestion,
+                                                                               "");
+                statsTable.setCollapsible(false);
+                
+                if (!statsTable.getQuestionStatisticsTable().isEmpty()) {
+                    statisticsTablesForTeam.add(statsTable);
+                }
+            }
+        
+            InstructorResultsQuestionTable.sortByQuestionNumber(statisticsTablesForTeam);
+            teamToStatisticsTables.put(team, statisticsTablesForTeam);
+        }
+        
+        sectionPanel.setTeamStatisticsTable(teamToStatisticsTables);
+    }
+
+    private void prepareHeadersForTeamPanelsInSectionPanel(
+                                    InstructorFeedbackResultsSectionPanel sectionPanel) {
+        switch (viewType) {
+            case GIVER_QUESTION_RECIPIENT:
+                sectionPanel.setStatisticsHeaderText("Statistics for Given Responses");
+                sectionPanel.setDetailedResponsesHeaderText("Detailed Responses");
+                break;
+            case RECIPIENT_QUESTION_GIVER:
+                sectionPanel.setStatisticsHeaderText("Received Responses Statistics");
+                sectionPanel.setDetailedResponsesHeaderText("Detailed Responses");
+                break;
+            default:
+                sectionPanel.setDetailedResponsesHeaderText("");
```

Comment:
Just wondering, but is there a reason why `detailedResponsesHeaderText` should have a default, but `statisticsHeaderText` doesn't?